src/paths/distances.c

```diff
@@ -777,3 +777,50 @@ igraph_error_t igraph_pseudo_diameter_dijkstra(const igraph_t *graph,
 
     return IGRAPH_SUCCESS;
 }
+
+/**
+ * \function igraph_graph_center
+ * \brief Centre vertices of a graph.
+ *
+ * The centre vertices of a graph is calculated by finding the vertices
+ * with the minimum eccentricity
+ *
+ * \param graph The input graph, it can be directed or undirected.
+ * \param res Pointer to an initialized vector, the result is stored
+ *    here.
+ * \param mode What kind of paths to consider for the calculation:
+ *    \c IGRAPH_OUT, paths that follow edge directions;
+ *    \c IGRAPH_IN, paths that follow the opposite directions; and
+ *    \c IGRAPH_ALL, paths that ignore edge directions. This argument
+ *    is ignored for undirected graphs.
+ * \return Error code.
+ *
+ * Time complexity: O(v*(|V|+|E|)), where |V| is the number of
+ * vertices, |E| is the number of edges and v is the number of
+ * vertices for which eccentricity is calculated.
+ *
+ * \sa \ref igraph_eccentricity().
+ *
+ */
+igraph_error_t igraph_graph_center(const igraph_t *graph, igraph_vector_t *res,
+                  igraph_neimode_t mode) {
+
+    igraph_vector_t ecc;
+    igraph_vs_t vids = igraph_vss_all();
+
+    IGRAPH_CHECK(igraph_vector_init(&ecc, 0));
```

Comment:
This vector is used only for the lifetime of the function. We need to ensure that it gets deallocated if an error is raised in the middle of the function. Check the documentation on error handling especially section 5.3, on how to do this:

https://igraph.org/c/html/develop/igraph-Error.html

You can also look at `igraph_regular_tree()` for a very simple example of how this is done.

We put the allocated vector on the "finally stack", along with its destructor, using an `IGRAPH_FINALLY` call. There are macros to simplify this for vectors. When the vector is destroyed below, we remove it from the finally stack using an `IGRAPH_FINALLY_CLEAN` call.